### ISSUE TITLE
Allow for creation of Vault IAM user and credentials 

### ISSUE THREAD

**Issue by hferentschik** [jenkins-x/terraform-aws-eks-jx#25]
As a workaround, until issue #20 is resolved and the vault_user becomes obsolete, we should improve the user experience around the required _vault_user_.

The intended improvement is two-fold. On the Terraform side we are going to create the required user if none is specified, something along the lines:

```terraform
resource "aws_iam_user" "test" {
  count = var.vault_user == "" ? 1 : 0

  name = "test-user"
}

resource "aws_iam_access_key" "test" {
  count = var.vault_user == "" ? 1 : 0

  user = aws_iam_user.test[0].name
}
```

We also add the name, is and secret to the output variables:

```terraform
output "vault_user" {
  value       = aws_iam_user.test[0].name
}

output "id" {
  value       = aws_iam_access_key.test[0].id
}

output "secret" {
  value       = aws_iam_access_key.test[0].secret
}
```

In the case where there is no _vault_user_ specified and the user has enough permissions to create new users, this will allow running boot like this:

```sh
$ VAULT_AWS_ACCESS_KEY_ID=$(terraform output id) VAULT_AWS_SECRET_ACCESS_KEY=$(terraform output secret) jx boot -r jx-requirements.yml 
```

The above will be documented with an additional note/info that in the case where new users cannot be created, a user needs to be provided and `jx boot` needs to be run as:

```sh
VAULT_AWS_ACCESS_KEY_ID=<id> VAULT_AWS_SECRET_ACCESS_KEY=<secret> jx boot -r jx-requirements.yml 
```

**Comment by zakswindow** [comment]
I think, this issue can be resolved if we resolve this https://github.com/jenkins-x/terraform-aws-eks-jx/issues/29.

**Comment by hferentschik** [comment]
Fixed by pull request #26
